src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -1109,7 +1110,7 @@ private List<ExternalProjectReference> GetProjectReferences(RemoteWalkContext co
                             _request.Project.Name,
                             proj.PackageSpecProjectName,
                             StringComparison.OrdinalIgnoreCase))
-                        .ToList();
+                        .AsList();
```

Comment:
This change is unnecessary. There's no functional change here. 
LINQ generates enumerates which had to be converted to List.